### PR TITLE
Correct and Add tests for TutorMatching screen

### PR DESCRIPTION
**Pull request by LeaBlancher**
> [!CAUTION]
>  TODO => PR description

Use waitForIdle to wait for the dialog to be show. Add tests for the filter and for the function isTutorAvailable.

### CODE REVIEW

**Review by aferrante1 — APPROVED** [review]
These tests suite does a great job of covering key edge cases and ensuring robustness.

File: app/src/test/java/com/github/se/project/ui/lesson/IsTutorAvailableTest.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.github.se.project.ui.lesson
+
+import org.junit.Test
+
+class IsTutorAvailableTest {
+  @Test
+  fun isTutorAvailable_respondToEdgeCases() {
+    // check if exception thrown when tutor schedule is empty
+    try {
```
[the remark this answers is not in the document]
Comment (LeaBlancher) [reply app/src/test/java/com/github/se/project/ui/lesson/IsTutorAvailableTest.kt:9]: Thank you for this remark, I knew a function exists and struggled to find it. I will updated that!

File: app/src/test/java/com/github/se/project/ui/lesson/IsTutorAvailableTest.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.github.se.project.ui.lesson
+
+import org.junit.Assert.assertThrows
+import org.junit.Test
+
+class IsTutorAvailableTest {
+  @Test
+  fun isTutorAvailable_respondToEdgeCases() {
+    // check if exception thrown when tutor schedule is empty
+    assertThrows(IllegalArgumentException::class.java) {
```
Comment (aferrante1) [inline app/src/test/java/com/github/se/project/ui/lesson/IsTutorAvailableTest.kt:10]: Great got changing the try catch to assertThrows!

**Review by aferrante1 — APPROVED** [review]
